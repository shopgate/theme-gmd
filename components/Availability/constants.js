/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * The availability states of a product.
 */
export const AVAILABILITY_STATE_OK = 'ok';
export const AVAILABILITY_STATE_WARNING = 'warning';
export const AVAILABILITY_STATE_ALERT = 'alert';
