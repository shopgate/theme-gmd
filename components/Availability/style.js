/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import colors from 'Styles/colors';

const stateOk = css({
  color: colors.success,
}).toString();

const stateWarning = css({
  color: colors.warning,
}).toString();

const stateAlert = css({
  color: colors.error,
}).toString();

export default {
  stateOk,
  stateWarning,
  stateAlert,
};
