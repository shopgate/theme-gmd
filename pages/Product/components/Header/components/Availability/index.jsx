/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import PlaceholderLabel from 'Components/PlaceholderLabel';
import AvailableText from 'Components/Availability';
import connect from './connector';
import styles from './style';

/**
 * The Availability component.
 * @param {Object} props The component props.
 * @return {JSX}
 */
const Availability = ({ availability }) => (
  <PlaceholderLabel className={styles.placeholder} ready={(availability !== null)}>
    {availability && (
      <AvailableText
        className={styles.availability}
        showWhenAvailable
        text={availability.text}
        state={availability.state}
      />
    )}
  </PlaceholderLabel>
);

Availability.propTypes = {
  availability: PropTypes.shape(),
};

Availability.defaultProps = {
  availability: null,
};

export default connect(Availability);
