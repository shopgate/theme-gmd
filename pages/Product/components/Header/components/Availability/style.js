/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';

const placeholder = css({
  height: 16,
  width: '70%',
  marginTop: 5,
  marginBottom: 2,
}).toString();

const availability = css({
  fontSize: '0.875rem',
}).toString();

export default {
  placeholder,
  availability,
};
