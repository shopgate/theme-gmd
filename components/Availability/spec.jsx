/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import {
  AVAILABILITY_STATE_OK,
  AVAILABILITY_STATE_WARNING,
  AVAILABILITY_STATE_ALERT,
} from './constants';

import Availability from './index';

describe('<Availability />', () => {
  it('should not render when text is empty', () => {
    const wrapper = mount(
      <Availability text="" />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should not render by default if state is "ok"', () => {
    const wrapper = mount(
      <Availability state={AVAILABILITY_STATE_OK} text="Available" />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render if state is "ok" when "showWhenAvailable" is set', () => {
    const wrapper = mount(
      <Availability state={AVAILABILITY_STATE_OK} text="Available" showWhenAvailable />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render if state is "warning"', () => {
    const wrapper = mount(
      <Availability state={AVAILABILITY_STATE_WARNING} text="Only 5 left" />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render if state is "warning"', () => {
    const wrapper = mount(
      <Availability state={AVAILABILITY_STATE_ALERT} text="Out of stock" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
