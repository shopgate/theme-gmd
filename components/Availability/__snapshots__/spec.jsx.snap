// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Availability /> should not render by default if state is "ok" 1`] = `
<Availability
  className=""
  showWhenAvailable={false}
  state="ok"
  text="Available"
/>
`;

exports[`<Availability /> should not render when text is empty 1`] = `
<Availability
  className=""
  showWhenAvailable={false}
  state="ok"
  text=""
/>
`;

exports[`<Availability /> should render if state is "ok" when "showWhenAvailable" is set 1`] = `
<Availability
  className=""
  showWhenAvailable={true}
  state="ok"
  text="Available"
>
  <div
    className=" css-1ixzglu"
  >
    Available
  </div>
</Availability>
`;

exports[`<Availability /> should render if state is "warning" 1`] = `
<Availability
  className=""
  showWhenAvailable={false}
  state="warning"
  text="Only 5 left"
>
  <div
    className=" css-14jedxl"
  >
    Only 5 left
  </div>
</Availability>
`;

exports[`<Availability /> should render if state is "warning" 2`] = `
<Availability
  className=""
  showWhenAvailable={false}
  state="alert"
  text="Out of stock"
>
  <div
    className=" css-qsvazi"
  >
    Out of stock
  </div>
</Availability>
`;
